README.md

```diff
@@ -1,16 +1,137 @@
+
+export AUTH_TYPE=CLI
+export KEYVAULT_NAME="YOUR_KEYVAULT_NAME"
+
+This command takes about 3-5 minutes:
+
+mvn clean package 
+
+mvn spring-boot:run
+
+# When the previous command shows 'Netty started on port(s): 8080' 
+# test the application (takes about 10 seconds to start) in a new window
+
+curl http://localhost:8080/healthz
+
+# Stop the application by typing Ctrl-C in the Azure CLI terminal window
+Or :::::
+mvn spring-boot:stop
+
+```
+
+## Run the application as a local container instead
+
+'''bash
+
+# Make sure you are in the root of the repo
+# Run the command below to stop the previous instance and free up port 8080:
+
+mvn spring-boot:stop 
+
+# You can also hit Ctrl-C from the az cli window
+
+# Build the image and run the container using Docker 
+#
+# make sure you are in the root of the repo
+
+# this will use Azure CLI cached credentials
+
+# build the image
+
+docker build -t helium-dev .
+
+# Then run the container.  
+
+# First obtain APPLICATIONINSIGHTS_CONNECTION_STRING using the following link:
+#TEST THIS
+https://docs.microsoft.com/en-us/azure/azure-monitor/app/java-get-started?tabs=maven
+
+export APP_INSIGHTS_CONNECTION_STRING=<KEY_FROM_ABOVE_LINK>
+
+docker run -p4120:4120 --env AUTH_TYPE=CLI --env KEYVAULT_NAME=$KEYVAULT_NAME --env APPLICATIONINSIGHTS_CONNECTION_STRING=$APP_INSIGHTS_CONNECTION_STRING -v ~/.azure:/home/helium/.azure helium-dev:latest
+
+# Check the logs to ensure the container is properly running
+# Re-run until the application started message appears
+docker ps -a (container id on left)
+docker logs container_id
+
+# test the application
+# the application takes about 10 seconds to start
+curl http://localhost:4120/healthz
+
+# Clean up  - stop and remove the container
+docker stop container_id
+
+```
 
 ## Contributing
 
 This project welcomes contributions and suggestions.  Most contributions require you to agree to a
 Contributor License Agreement (CLA) declaring that you have the right to, and actually do, grant us
 the rights to use your contribution. For details, visit [Microsoft Contributor License Agreement](https://cla.opensource.microsoft.com).
 
-When you submit a pull request, a CLA-bot will automatically determine whether you need to provide
-a CLA and decorate the PR appropriately (e.g., label, comment). Simply follow the instructions
+When you submit a pull request, a CLA bot will automatically determine whether you need to provide
+a CLA and decorate the PR appropriately (e.g., status check, comment). Simply follow the instructions
 provided by the bot. You will only need to do this once across all repos using our CLA.
 
 This project has adopted the [Microsoft Open Source Code of Conduct](https://opensource.microsoft.com/codeofconduct/).
 For more information see the [Code of Conduct FAQ](https://opensource.microsoft.com/codeofconduct/faq/) or
 contact [opencode@microsoft.com](mailto:opencode@microsoft.com) with any additional questions or comments.
+
```

Comment:
There are a lot of comments in here and it's getting confusing.  Let's address all of the comments, close the PR and create a clean PR so we don't miss anything.

Reply:
I'll do another PR for issues 141 175 190 and 73.  As I kept doing this it kept growing and other new and old issues were linked.